internal/dag/cache.go

```diff
@@ -553,6 +572,25 @@ func (kc *KubernetesCache) secretTriggersRebuild(secret *v1.Secret) bool {
 		}
 	}
 
+	if kc.gateway != nil {
+		for _, listener := range kc.gateway.Spec.Listeners {
+			if listener.TLS == nil {
+				continue
+			}
+			if listener.TLS.CertificateRef == nil {
+				continue
+			}
+
+			ref := listener.TLS.CertificateRef
+			if ref.Kind == "Secret" &&
+				(ref.Group == "core" || ref.Group == "v1") {
```

Comment:
I can take it out. It's not clear if it should be `v1`, `core`, or `""`.

Reply:
Based upon validation it can't be empty string, so that just leaves `core`. I'll just update and we'll address later on if there's confusion from users.